### PR TITLE
Add Cyclomatic Complexity implementation for Java

### PR DESCRIPTION
**Pull request by marco-ballario**
This PR adds the Cyclomatic Complexity metrics implementation for the Java language. Two unit tests are added to verify the correct computation of the measures. Thanks in advance for the review.

### CODE REVIEW

File: src/metrics/cyclomatic.rs [hunk 1]
```diff
@@ -203,7 +203,19 @@ impl Cyclomatic for CppCode {
...
 impl Cyclomatic for PreprocCode {}
 impl Cyclomatic for CcommentCode {}
-impl Cyclomatic for JavaCode {}
+
+impl Cyclomatic for JavaCode {
+    fn compute(node: &Node, stats: &mut Stats) {
+        use Java::*;
+
+        match node.object().kind_id().into() {
+            If | For | While | Case | Catch | TernaryExpression | AMPAMP | PIPEPIPE => {
```
Comment (calixteman) [inline src/metrics/cyclomatic.rs:212]: It'd be nice in your tests to cover all the possibilities (there are not so much).

Comment (marco-ballario) [reply src/metrics/cyclomatic.rs:212]: Thank you for the advice! I extended my tests to cover all the different possibilities.

**Comment by dburriss** [comment]
Nice good to see.

**Review by calixteman — APPROVED** [review]
LGTM. Thank you.

**Comment by calixteman** [comment]
Unrelated to this change, but in looking to the code for js, I think we (I ??) missed to add operators like `??` and `?.` (maybe others ?): they're lazy evaluated like a `||` or a `&&`.
Oh and probably `||=`,  `&&=` are missing too.

**Comment by Luni-4** [comment]
Yep, right, we forgot them. Could you please open an issue in order to remember them? Thank you @calixteman! :)

**Pull request merged**